### DISCUSSION TITLE
Tidak Bisa Mengirim Chat menggunakan PHP 5.6 CURL

### DISCUSSION THREAD

**Discussion by lukman-bahar-ap** [kirimwa-id/forum#93] [Tanya Jawab] [answered]
mohon bantuan saya tidak bisa mengirim chat, sebelumnya bisa menggunakan kode ini, saya masih menggunakan php versi 5.6

Error : 
> {"message":"Bad JSON input."}

Code :

> function sendWaMessage($msg, $device, $destination,  $apitoken){
>
>  $url = "https://api.kirimwa.id/v1/messages";
>  $paramspayload = array(
>    "device_id" => $device,
>    "message" => $msg,
>    "phone_number" => $destination,
>    "message_type" => "text");
>  
>  $authorization = "Authorization: Bearer ".$apitoken;
>
>  $ch = curl_init($url);
>  curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, 0);
>  curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, 0);
>  curl_setopt($ch, CURLOPT_POST, 1);
>  curl_setopt($ch, CURLOPT_POSTFIELDS, $paramspayload); 
>  curl_setopt($ch, CURLOPT_HTTPHEADER, array('Content-Type: application/json' , $authorization));
>  curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
>  curl_setopt($ch, CURLOPT_FOLLOWLOCATION, 1);
>  $result = curl_exec($ch);
>  curl_close($ch);
>  print_r ($result);
>
> }

**Comment by rioastamal** [comment, marked as answer]
@lukman-bahar-ap itu berarti JSON yang dikirimkan tidak valid. Pastikan JSON dalam variabel `$msg` sudah valid JSON syntax. Jika nilai dari `$msg` adalah dinamis maka coba `echo` terlebih dahulu nilainya lalu verify dengan JSON parser misal di `jq` atau online JSON parser misal http://json.parser.online.fr/

**Reply by lukman-bahar-ap** [reply]
Alhamdulillah, sudah di coba hardcode. butuh  *json_encode* untuk parameter array
>   curl_setopt($ch, CURLOPT_POST, 1);
>   curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($paramspayload)); 

Terima kasih bantuannya mas @rioastamal
